pandas/plotting/_matplotlib/hist.py

```diff
@@ -417,11 +419,21 @@ def hist_frame(
         if not isinstance(column, (list, np.ndarray, ABCIndexClass)):
             column = [column]
         data = data[column]
-    data = data._get_numeric_data()
+    # GH32590
+    columns_copy = data.columns
+    numeric_data = data._get_numeric_data()
+    datetime_data = data.select_dtypes(include="datetime64[ns]")
+    data = concat([numeric_data, datetime_data], axis=1)
     naxes = len(data.columns)
 
     if naxes == 0:
-        raise ValueError("hist method requires numerical columns, nothing to plot.")
+        raise ValueError(
+            "hist method requires numerical or datetime columns, nothing to plot."
+        )
+    else:
+        data = data.reindex(
+            columns=[name for name in columns_copy if name in data.columns]
```

Comment:
you can pass `copy=False`

Reply:
The `else` condition has two purposes.
The first point is `columns_copy` may not equal to new `data.columns` if there's any non numerical or datetime data.
`else` condition also keeps the original order of columns because `concat` may destroy the order.
So I think the `else` condition is needed here.